controllers/feeds/reactionList.js

```diff
@@ -63,3 +59,31 @@ module.exports = async (req, res) => {
     });
   }
 };
+
+function sortCommentsCreatedAtAscending(comments) {
+  return [...comments].sort(
+    (prev, next) => moment(prev.created_at).unix() - moment(next.created_at).unix()
+  );
+}
+
+function buildComment(level, comments) {
+  if (!Array.isArray(comments)) return [];
+
+  const built = comments?.map((comment) => {
+    const nextLevelComments = buildComment(level + 1, comment?.latest_children?.comment);
+
+    if (level === MAX_COMMENT_LEVEL)
+      return {
+        ...comment
+      };
+
+    return {
+      ...comment,
+      latest_children: {
+        comment: sortCommentsCreatedAtAscending(nextLevelComments)
+      }
+    };
+  });
+
+  return sortCommentsCreatedAtAscending(built);
+}
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `sortCommentsCreatedAtAscending` and `buildComment` functions are missing error handling. If there's an issue with the input data, these functions could throw errors which aren't currently being caught. Consider adding error handling to these functions.
```diff
function sortCommentsCreatedAtAscending(comments) {
+   if (!Array.isArray(comments)) return [];
  return [...comments].sort(
    (prev, next) => moment(prev.created_at).unix() - moment(next.created_at).unix()
  );
}

function buildComment(level, comments) {
  if (!Array.isArray(comments)) return [];

  const built = comments?.map((comment) => {
    const nextLevelComments = buildComment(level + 1, comment?.latest_children?.comment);

    if (level === MAX_COMMENT_LEVEL)
      return {
        ...comment
      };

    return {
      ...comment,
      latest_children: {
        comment: sortCommentsCreatedAtAscending(nextLevelComments)
      }
    };
  });

  return sortCommentsCreatedAtAscending(built);
}
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->